src/backend/utils/activity/pgstat_function.c

```diff
@@ -68,10 +68,48 @@ pgstat_drop_function(Oid proid)
 							  proid);
 }
 
+/*
+ * Wrapper function that calls the initilization function.
+ * Calls the pre function call hook on the procname 
+ * before invoking the initilization function. Performing a 
+ * system cache search in case fcinfo isnull for getting the procname
+ */
+
+void
+pgstat_init_function_usage_wrapper(FunctionCallInfo fcinfo,
+						   PgStat_FunctionCallUsage *fcusageptr, char *procname)
+{
+	
+	if (pre_function_call_hook && IsTransactionState())
```

Comment:
If we are moving the entire function into TSQL, we wont require pre_function_call_hook anymore, check for its validity and scrap it